### PR TITLE
Use  `vitest-browser-react` and `browserUserEvent`

### PR DESCRIPTION
**Pull request by amanmahajan7**
https://github.com/vitest-dev/vitest-browser-react

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -707,7 +707,7 @@ export default [
       'testing-library/prefer-presence-queries': 1,
       'testing-library/prefer-query-by-disappearance': 1,
       'testing-library/prefer-query-matchers': 0,
-      'testing-library/prefer-screen-queries': 1,
+      'testing-library/prefer-screen-queries': 0,
```
Comment (amanmahajan7) [inline eslint.config.js:710]: Had to disable it as complains about `page...`

File: test/browser/TextEditor.test.tsx [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 }
 
-const columns: readonly Column<Row>[] = [{ key: 'name', name: 'Name', renderEditCell: textEditor }];
+const columns: readonly Column<Row>[] = [
+  {
+    key: 'name',
+    name: 'Name',
+    renderEditCell: textEditor,
+    editorOptions: {
+      commitOnOutsideClick: false
```
Comment (amanmahajan7) [inline test/browser/TextEditor.test.tsx:18]: Added this so we can test the blur event

File: test/browser/column/renderEditCell.test.tsx [hunk 3]
```diff
@@ -14,79 +14,78 @@ interface Row {
...
+    const saveButton = page.getByRole('button', { name: 'save' });
 
     // await userEvent.click() triggers both mousedown and click, but without delay,
     // which isn't realistic, and isn't enough to trigger outside click detection
-    await user.pointer([{ keys: '[MouseLeft>]', target: saveButton }]);
-    await act(async () => {
-      await new Promise(requestAnimationFrame);
-    });
-    await user.pointer({ keys: '[/MouseLeft]' });
+    await userEvent.click(saveButton, { delay: 10 });
```
Comment (amanmahajan7) [inline test/browser/column/renderEditCell.test.tsx:84]: 😎

File: test/browser/column/renderEditCell.test.tsx [hunk 4]
```diff
@@ -275,31 +274,30 @@ describe('Editor', () => {
...
-      expect(col1Input).toHaveFocus();
+      const col1Input = page.getByRole('textbox', { name: 'col1-input' });
+      await expect.element(col1Input).toHaveFocus();
       await userEvent.click(outerInput);
-      expect(outerInput).toHaveFocus();
+      await expect.element(outerInput).toHaveFocus();
       await waitFor(() => {
-        expect(col1Input).not.toBeInTheDocument();
+        expect(col1Input.query()).not.toBeInTheDocument();
       });
```
Comment (amanmahajan7) [inline test/browser/column/renderEditCell.test.tsx:292]: Need a better way

File: test/browser/column/renderEditCell.test.tsx [hunk 5]
```diff
@@ -14,79 +14,77 @@ interface Row {
...
-    await user.keyboard('234');
+    page.render(<EditorTest onSave={onSave} />);
+    await userEvent.dblClick(getCellsAtRowIndex(0)[0]);
+    await userEvent.keyboard('234');
     expect(onSave).not.toHaveBeenCalled();
-    const saveButton = screen.getByRole('button', { name: 'save' });
+    const saveButton = page.getByRole('button', { name: 'save' });
 
     // await userEvent.click() triggers both mousedown and click, but without delay,
     // which isn't realistic, and isn't enough to trigger outside click detection
```
[the remark this answers is not in the document]
Comment (amanmahajan7) [reply test/browser/column/renderEditCell.test.tsx:82]: I think it is true in this case also, no?
https://playwright.dev/docs/api/class-locator#locator-click-option-delay

File: test/browser/column/renderEditCell.test.tsx [hunk 6]
```diff
@@ -97,101 +95,100 @@ describe('Editor', () => {
...
 
     it('should not be editable if editable is false', async () => {
-      render(<EditorTest editable={false} />);
+      page.render(<EditorTest editable={false} />);
       const cell = getCellsAtRowIndex(0)[1];
       expect(cell).toHaveAttribute('aria-readonly', 'true');
       await userEvent.dblClick(cell);
-      expect(screen.queryByRole('textbox', { name: 'col2-editor' })).not.toBeInTheDocument();
+      // eslint-disable-next-line testing-library/prefer-presence-queries
+      expect(page.getByRole('textbox', { name: 'col2-editor' }).query()).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (amanmahajan7) [reply test/browser/column/renderEditCell.test.tsx:133]: `element` throws an error if the textbox is not in the DOM

File: test/browser/column/renderEditCell.test.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (nstepien) [reply test/browser/column/renderEditCell.test.tsx:82]: Just saying we should document why the delay is necessary here

File: test/browser/column/renderEditCell.test.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (nstepien) [reply test/browser/column/renderEditCell.test.tsx:133]:
```ts
await expect.element(page.getByRole('random')).not.toBeInTheDocument();
```

tested and this works fine 🤔

[the remark this answers is not in the document]
Comment (amanmahajan7) [reply test/browser/column/renderEditCell.test.tsx:133]: It does work. I remember seeing an error 😕

[the remark this answers is not in the document]
Comment (amanmahajan7) [reply test/browser/column/renderEditCell.test.tsx]: May be I was using https://vitest.dev/guide/browser/locators.html#element

File: test/browser/column/renderEditCell.test.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (nstepien) [reply test/browser/column/renderEditCell.test.tsx:79]:
```suggestion
    // without delay, `click()` triggers both `mousedown` and `click` events
    // too quickly for outside clicks to be detected
```

**Review by nstepien — APPROVED** [review]

**Pull request merged**